scripts/create-local-tag-for-release

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Script to create a new local tag in preparation for a release
+
+set -euo pipefail
+
+REPO_ROOT_PATH="$( cd "$(dirname "$0")"; cd ../; pwd -P )"
+MAKEFILE_PATH=$REPO_ROOT_PATH/Makefile
+LATEST_TAG=$(make -s -f $MAKEFILE_PATH latest-release-tag | cut -b 2- )
+
+HELP=$(cat << 'EOM'
+  Create a new local tag in preparation for a release.
+
+  Usage: create-local-tag-for-release [options]
+
+  Options:
+    -v      new tag / version number. The script relies on the user to specify a valid and accurately incremented tag.
+    -m      increment major version
+    -i      increment minor version
+    -p      increment patch version
+    -h      help
+
+  Examples:
+    create-local-tag-for-release -v 1.0.0       Create local tag for new version 1.0.0
+    create-local-tag-for-release -i             Create local tag for new version by incrementing minor version only (previous tag=1.0.0, new tag=1.1.0)
+EOM
+)
+
+MAJOR_INC=false
+MINOR_INC=false
+PATCH_INC=false
+NEW_TAG=""
+
+process_args() {
+    while getopts "hmipv:" opt; do
+        case ${opt} in
+            h )
+            echo -e "$HELP" 1>&2
+            exit 0
+            ;;
+            m )
+            MAJOR_INC=true
+            ;;
+            i )
+            MINOR_INC=true
+            ;;
+            p )
+            PATCH_INC=true
+            ;;
+            v )
+            NEW_TAG="${OPTARG}"
+            ;;
+            \? )
+            echo "$HELP" 1>&2
+            exit 0
+            ;;
+        esac
+    done
+}
+
+validate_args() {
+    if [[ ! -z $NEW_TAG ]]; then
+        echo "🥑 Using the new tag specified with -v flag. All other flags, if specified, will be ignored."
+        return
+    fi
+
+    if ($MAJOR_INC && $MINOR_INC) || ($MAJOR_INC && $PATCH_INC) || ($MINOR_INC && $PATCH_INC); then
+        echo "❌ Invalid arguments passed. Specify only one of 3 tag parts to increment for the new tag: -m (major) or -i (minor) or -p (patch)."
+        exit 1
+    fi
+
+    if $MAJOR_INC || $MINOR_INC || $PATCH_INC; then
+        return
+    fi
+
+    echo -e "❌ Invalid arguments passed. Specify atleast one argument.\n$HELP"
+    exit 1
+}
+
+# create the new tag
+create_tag() {
+    git tag v$NEW_TAG
```

Comment:
should there be some kind of regex validation here?

Reply:
sure, I can add regex to cover the case of typos